### PR TITLE
endpoint: Do not fail to regenerate endpoint when running w/o L7 proxy

### LINKED ISSUE #9518: Reject CNP with L7 rules when L7 proxy is disabled
#9481 will prevent endpoint regeneration from failing when L7 proxy is disabled. This is non ideal, as it will let users to think that an L7 policy was enforced when the L7 proxy was not running.

To avoid this, we should log a warning in the endpoint regeneration, and also reject CNPs with L7 rules.

### PR DESCRIPTION
**Pull request by brb**
It is expected that in some deployments the L7 proxy will be disabled. Currently, it can be disabled when `--install-iptables-rules=false` is set which can happen when e.g.  running in the ipvlan mode or
w/o netfilter/iptables.

To play nicely with such deployments, do not fail the endpoint regeneration when the L7 proxy is disabled.

Fixes #9518

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9481)

### CODE REVIEW

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-me-please

**Comment by ianvernon** [comment]
I don't think this is a complete enough fix. We should disable L7 policy being allowed to be imported via Cilium in this case as well. Otherwise we will tell users they are realizing policy which actually isn't being realized. It's not clear at first that disabling iptables rules would disable the proxy, which in turn disables an entire layer of policy enforcement.

**Review by aanm — APPROVED** [review]

**Comment by ianvernon** [comment]
> I agree that it's very unclear. What if I introduce --enable-l7-proxy (default "true"), and log a fatal message when --enable-l7-proxy=true and --install-iptables-rules=false?

Works for me.

> AFAIK, the policy import happens async, so in the best case we could just log an error? I'd like to keep such change out of the scope of this PR.

Please log a warning if there are L7 rules in the policy and we have the proxy disabled.

**Review by ianvernon — CHANGES_REQUESTED** [review]
Commented out of band of the review.

**Comment by brb** [comment]
test-me-please

**Review by joestringer — APPROVED** [review]
The changes that are here are simple and straightforward.

I understand from the community meeting that there is a desire to also reject CNPs that configure L7 rules (and perhaps also FQDN rules) in this PR as well.

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
PTAL. Created a separate issue with the `release-blocker` priority to track warning/rejecting CNPs with L7 rules and `--enable-l7-proxy=false`: https://github.com/cilium/cilium/issues/9518.

File: install/kubernetes/cilium/charts/config/templates/configmap.yaml [hunk 1]
```diff
@@ -193,6 +193,12 @@ data:
   flannel-manage-existing-containers: "{{ .Values.global.flannel.manageExistingContainers }}"
 {{- end }}
 
+{{- if .Values.global.l7Proxy }}
+
```
Comment (aanm) [inline install/kubernetes/cilium/charts/config/templates/configmap.yaml:197]: remove new line?

**Review by ianvernon — APPROVED** [review]

**Comment by brb** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

File: daemon/daemon.go [hunk 2]
```diff
@@ -433,11 +433,11 @@ func NewDaemon(ctx context.Context, dp datapath.Datapath) (*Daemon, *endpointRes
 	// daemon's proxy into each endpoint.
 	bootstrapStats.proxyStart.Start()
 	// FIXME: Make the port range configurable.
-	if option.Config.InstallIptRules {
+	if option.Config.EnableL7Proxy {
 		d.l7Proxy = proxy.StartProxySupport(10000, 20000, option.Config.RunDir,
 			option.Config.AccessLog, &d, option.Config.AgentLabels, d.datapath, d.endpointManager)
 	} else {
-		log.Warning("L7 proxies not supported when --install-iptables-rules=\"false\"")
+		log.Info("L7 proxies is disabled")
```
Comment (joestringer) [inline daemon/daemon.go]: nit:
```suggestion
		log.Info("L7 proxies are disabled")
```

**Review by joestringer — APPROVED** [review]
Minor nit.

**Comment by aanm** [comment]
test-me-please

**Pull request merged**